### PR TITLE
Cyril/frontend migration

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR handles the general migration for all components

## Type of change
- [x] New feature (non-breaking change which adds functionality)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [x] Login Screen
- [x] Single Upload Page
- [x] Batch Upload Page
- [x] All Projects Page
- [x] Sample Details Page
- [x] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

### Single Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

### Batch Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from the previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

### All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

### Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

### Report Page
* Data loads successfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [x] I have run the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,7 +1,14 @@
-<div>
+<div id='user_create_page'>
   <% if current_user && current_user.admin %>
-    <%= react_component 'CreateUser', { csrf: form_authenticity_token, user: @user } %>
+    <%= javascript_tag do %>
+      react_component('CreateUser', {
+        user: JSON.parse('<%= raw escape_json(@user) %>'),
```
Comment (cdebourcy) [inline app/views/users/new.html.erb]: Have you tried the json_escape function Trevor suggested? Did it work? Then you shouldn't need the custom function escape_json anymore

Comment (cyrielo) [reply app/views/users/new.html.erb]: Oh I have tried it, it escaped the json using a single backslash, so I update the custom method we created to use the json_escape method trevor suggested, but to replace single backslash with double instead

File: app/helpers/application_helper.rb [hunk 2]
```diff
@@ -13,4 +13,11 @@ def hash_array_json2csv(input_file, output_file, keys)
       end
     end
   end
+
+  def escape_json(hash)
+    # using json_escape to prevent XSS vulnerability
+    str = json_escape(hash.to_json) unless hash.class == 'String'
+    str = str.gsub!("\\", "\\\\\\") if str.include? "\\"
```
Comment (trevorb) [inline app/helpers/application_helper.rb]: Strings need to be json_escaped as well, I'd recommend that you use json_escape like the docs recommend rather than JSON.parse on the output.

https://apidock.com/rails/ERB/Util/json_escape
